pkg/status/statuscollector.go

```diff
@@ -41,7 +44,19 @@ func (c *Controller) syncStatusCollector(ctx context.Context, ref string) error
 		statusCollector.Name = ref
 	}
 
-	// TODO: validate the statusCollector if not nil
+	// Validate the StatusCollector
+	if errs := c.validateStatusCollector(statusCollector); len(errs) > 0 {
+		if err := c.updateStatusCollectorErrors(ctx, statusCollector, errs); err != nil {
```

Comment:
It might be worth putting a comment on the variable `statusCollector` about its immutability (see https://github.com/kubestellar/kubestellar/blob/08d290a0dbcfe91714d7e140cd80630c5ed843b9/pkg/generated/listers/control/v1alpha1/statuscollector.go#L36).